vendor/arbiter/arbiter.cpp

```diff
@@ -1386,15 +1402,38 @@ std::unique_ptr<Http> Http::create(Pool& pool)
 }
 
 std::unique_ptr<std::size_t> Http::tryGetSize(std::string path) const
+{
+    return tryGetSize(path, http::Headers());
+}
+
+std::size_t Http::getSize(
```

Comment:
It seems like this is new. Is there some reason for it to be added that I'm missing?

Reply:
I'm not sure it's quite "new" as it's the same code body as before.  Just with missing overloads added (which already exist for the get/put functions, just not for the HEAD request calls) which forward the query/header params.  Otherwise you get a 401 error every time you hit [this line](https://github.com/PDAL/PDAL/pull/3439/files#diff-642075337fbe162193f18e3871532a17abb3e333a7d0bfedd57a66fb905b07dbR686) if your EPT endpoint is authenticated with a query or header parameter value.

Reply:
Hmm. On my diff I don't see any `getSize()` in the old stuff, just `tryGetSize()`. I wouldn't care except this throws.

Reply:
Yes that's true, this specific overload of `getSize` is new - it's to bring the header/query-forwarding versions of get/getBinary/getSize/tryGet/tryGetBinary/tryGetSize to parity with the normal versions.

The generic `Driver` has [all of these](https://github.com/PDAL/PDAL/blob/master/vendor/arbiter/arbiter.hpp#L3751-L3781), which is what I mean by this function not being "new", per se.

The HTTP version is intended to have overloads of each of these which have the same functionality but also forward query/header params, but the `size` ones were missing from [here](https://github.com/PDAL/PDAL/blob/master/vendor/arbiter/arbiter.hpp#L4042-L4072).

So I added those overloads (the `getSize` one is not actually used, but now exists for consistency) and then updated the `getLocalHandle` function, which accepts query/header params, to use this new forwarding `tryGetSize` rather than omitting these params.